extension-iap/src/iap_android.cpp

```diff
@@ -68,36 +70,46 @@ static int IAP_List(lua_State* L)
     env->DeleteLocalRef(products);
 
     free(buf);
-    assert(top == lua_gettop(L));
     return 0;
 }
 
 static int IAP_Buy(lua_State* L)
 {
-    int top = lua_gettop(L);
+    DM_LUA_STACK_CHECK(L, 0);
 
+    int top = lua_gettop(L);
     const char* id = luaL_checkstring(L, 1);
+    const char* token = "";
+
+    if (top >= 2 && lua_istable(L, 2)) {
+        luaL_checktype(L, 2, LUA_TTABLE);
+        lua_pushvalue(L, 2);
+        lua_getfield(L, -1, "token");
```

Comment:
What this token is? Should we add it in doc and script_api?

Reply:
Added. It's a subscription specific thing to distinguish between different offers that the user is eligible for.

https://developer.android.com/reference/com/android/billingclient/api/BillingFlowParams.ProductDetailsParams.Builder#setOfferToken(java.lang.String)